src/storage.rs

```diff
@@ -174,26 +204,36 @@ where
 
     // @TODO specify more useful error type
     fn init_from_existing(&mut self, files: Vec<DirEntry>) -> Result<(), ()> {
-        self.blobs = files
+        let mut runtime = tokio::runtime::Runtime::new().unwrap();
+        let mut max_blob_file_index: Option<usize> = None;
+        let mut active_blob = None;
+        let temp_blobs = files
             .iter()
-            .filter_map(|entry| {
-                // @TODO implement more file validations
-                if entry.metadata().ok()?.is_file() {
-                    Some(Blob::from_file(entry.path()).ok()?)
+            .map(DirEntry::path)
+            .filter(|path| path.is_file())
+            .filter_map(|path| {
+                let blob = runtime.block_on(Blob::from_file(path.clone())).unwrap();
```

Comment:
Opening a BLOB should happen only after you have successfully parsed its file name!

Reply:
outdated